au/code/au/operators.hh

```diff
@@ -37,57 +35,57 @@ namespace detail {
 //
 
 struct Equal {
-    template <typename T>
-    constexpr bool operator()(const T &a, const T &b) const {
+    template <typename T, typename U>
```

Comment:
So what necessitated this change?  Are we now comparing two different types?

Reply:
For now, we're just _paving the way_ to compare two different types.  We'll need this when we end up passing, say, a signed and unsigned value together to a comparison function.  But in the present PR, this is just a no-op refactor.

Reply:
Makes sense, just making sure I wasn't missing something.